tree/ntuple/v7/src/RNTupleModel.cxx

```diff
@@ -22,8 +22,109 @@
 #include <atomic>
 #include <cstdlib>
 #include <memory>
+#include <typeinfo>
 #include <utility>
 
+ROOT::Experimental::RResult<void>
+ROOT::Experimental::RNTupleModel::RProjectedFields::EnsureValidMapping(const Detail::RFieldBase *target,
+                                                                       const FieldMap_t &fieldMap)
+{
+   auto source = fieldMap.at(target);
+   if (source->GetStructure() != target->GetStructure())
+      return R__FAIL("field mapping structural mismatch: " + source->GetName() + " --> " + target->GetName());
+   if (target->GetStructure() == ENTupleStructure::kLeaf) {
+      if (typeid(*target) != typeid(*source))
```

Comment:
Maybe use `RFieldBase::GetType()` instead of RTTI?

Reply:
[done](https://github.com/root-project/root/pull/11973/commits/6260749392268713d051560708144f84f912cb58)